### PR TITLE
Wrap BaseSSLError into SSLError during response reads

### LINKED ISSUE #1764: SSLError
urllib 1.25.6
```
Traceback (most recent call last):
  File "/home/web/web/yandex/.env/lib/python3.5/site-packages/ioweb/network_service.py", line 177, in thread_network
    req, res, error, raise_network_error=False
  File "/home/web/web/yandex/.env/lib/python3.5/site-packages/ioweb/transport.py", line 310, in prepare_response
    self.read_with_timeout(req, res)
  File "/home/web/web/yandex/.env/lib/python3.5/site-packages/ioweb/transport.py", line 266, in read_with_timeout
    chunk = self.urllib3_response.read(chunk_size)
  File "/home/web/web/yandex/.env/lib/python3.5/site-packages/urllib3/response.py", line 507, in read
    data = self._fp.read(amt) if not fp_closed else b""
  File "/usr/lib/python3.5/http/client.py", line 448, in read
    n = self.readinto(b)
  File "/usr/lib/python3.5/http/client.py", line 478, in readinto
    return self._readinto_chunked(b)
  File "/usr/lib/python3.5/http/client.py", line 573, in _readinto_chunked
    chunk_left = self._get_chunk_left()
  File "/usr/lib/python3.5/http/client.py", line 541, in _get_chunk_left
    chunk_left = self._read_next_chunk_size()
  File "/usr/lib/python3.5/http/client.py", line 501, in _read_next_chunk_size
    line = self.fp.readline(_MAXLINE + 1)
  File "/usr/lib/python3.5/socket.py", line 576, in readinto
    return self._sock.recv_into(b)
  File "/home/web/web/yandex/.env/lib/python3.5/site-packages/urllib3/contrib/pyopenssl.py", line 332, in recv_into
    raise ssl.SSLError("read error: %r" % e)
ssl.SSLError: ("read error: Error([('SSL routines', 'ssl3_get_record', 'decryption failed or bad record mac')],)",)
```
Is it OK?
Should not it be `exceptions.SSLError` or something like that derived from urllib3 package?

### PR DESCRIPTION
**Pull request by sethmlarson**
Closes #1764

### CODE REVIEW

**Review by pquentin — APPROVED** [review]
I guess writing a test for this is going to be complicated and not worth it?

**Comment by sethmlarson** [comment]
I'll create a small test case :)

File: src/urllib3/response.py [hunk 1]
```diff
@@ -443,9 +444,8 @@ def _error_catcher(self):
             except BaseSSLError as e:
                 # FIXME: Is there a better way to differentiate between SSLErrors?
                 if "read operation timed out" not in str(e):  # Defensive:
```
Comment (pquentin) [inline src/urllib3/response.py:446]: Can we remove that Defensive pragma? Without that, I think we won't get coverage information for that condition.

Comment (sethmlarson) [reply src/urllib3/response.py:446]: Oh yeah, I'll remove that :) Nice catch!

**Pull request merged**